### PR TITLE
fix email subject for application create/approve notifications

### PR DESCRIPTION
**Pull request by pkarman**
Fixes bug in email subject where I18n key was wrong, and adds tests to verify.

### CODE REVIEW

File: spec/requests/applications_spec.rb [hunk 1]
```diff
@@ -24,13 +24,35 @@
...
+      login_as(user)
+
+      post users_applications_path, application: { name: 'test' }
+
+      app = Application.last
+      expect(response.status).to eq(302)
+      expect(deliveries.count).to eq(2)
+      expect(deliveries[0].subject).to eq(
+        I18n.t('dashboard.mailer.new_application.subject', id: app.issuer)
+      )
```
Comment (monfresh) [inline spec/requests/applications_spec.rb]: Should we also add an expectation that verifies that the email is sent to the admin (as opposed to the user, for example)?

File: spec/requests/applications_spec.rb [hunk 2]
```diff
@@ -24,13 +24,35 @@
...
 
       put users_application_path(app), application: { approved: 'true' }
 
       expect(response.status).to eq(302) # redirect on success
       app.reload
       expect(app.approved?).to eq(true)
+      expect(deliveries.count).to eq(2)
+      expect(deliveries[0].subject).to eq(
+        I18n.t('dashboard.mailer.approved_application.subject', id: app.issuer)
+      )
```
Comment (monfresh) [inline spec/requests/applications_spec.rb]: Same comment here about verifying the recipient.

**Comment by pkarman** [comment]
good call, @monfresh. I had to think a bit about the best way to test that, since the emails are not in a predictable order. See if this latest commit helps.

**Comment by monfresh** [comment]
Hmm. I left a comment here yesterday about only testing that the appropriate UserMailer is called in the request specs, and then adding mailers specs to test the actual mailer, but I don't see it anymore. Did you receive it?

**Comment by pkarman** [comment]
@monfresh I did not see that comment. Odd.

I could restructure the tests that way, which would necessitate refactoring where/how the UserMailer is used in the controller. Happy to do that.

**Comment by monfresh** [comment]
You don't need to change your controller. Here is how you can test that the correct mailers are being called:
```ruby
describe 'notifications' do
  it 'sends email to admin requesting approval' do
    ClimateControl.modify ADMIN_EMAIL: 'identity-admin@example.com' do
      user = create(:user)
      deliveries.clear
      login_as(user)

      mailer = instance_double(ActionMailer::MessageDelivery)
      allow(UserMailer).to receive(:admin_new_application).with(anything).and_return(mailer)
      allow(UserMailer).to receive(:user_new_application).with(anything).and_return(mailer)
      allow(mailer).to receive(:deliver_later)

      post users_applications_path, application: { name: 'test' }

      expect(UserMailer).to have_received(:admin_new_application).with(Application.last)
      expect(UserMailer).to have_received(:user_new_application).with(Application.last)
      expect(mailer).to have_received(:deliver_later).twice

      expect(response.status).to eq(302)
    end
  end
end
```

**Comment by monfresh** [comment]
Then your mailer specs would look something like this: https://github.com/18F/identity-idp/blob/master/spec/mailers/user_mailer_spec.rb

**Comment by pkarman** [comment]
@monfresh thanks for the detailed examples. I have committed those changes.

File: spec/requests/applications_spec.rb [hunk 3]
```diff
@@ -22,15 +22,52 @@
...
+        app = Application.last
+
+        expect(UserMailer).to have_received(:admin_new_application).with(app)
+        expect(UserMailer).to have_received(:user_new_application).with(app)
+        expect(mailer).to have_received(:deliver_later).twice
 
-      expect(response.status).to eq(302) # redirect on success
-      app.reload
-      expect(app.approved?).to eq(true)
+        expect(response.status).to eq(302)
```
Comment (monfresh) [inline spec/requests/applications_spec.rb]: Looks great! Perhaps add another expectation here to make sure the app is not yet approved?

**Comment by pkarman** [comment]
@monfresh done

**Comment by monfresh** [comment]
Looks good. Wanna squash?

**Comment by pkarman** [comment]
:squash:'d

**Pull request merged**